bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/internal/IconServlet.java

```diff
@@ -97,21 +102,62 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             return;
         }
 
-        String filename = StringUtils.substringAfterLast(req.getRequestURI(), "/");
-
-        String iconName = StringUtils.substringBeforeLast(filename, ".");
+        String category = getCategory(req);
+        Format format = getFormat(req);
```

Comment:
OK, that means if an icon set wants to support "svg" all icons has be SVG, right? But if I want to override it with a custom Icon, does it have to be SVG, too? From the UI I don´t know if it will return a custom icon or not, so I will always request it with svg.